File: Exercism.io/python/bob/bob.py

def is_question(what):
    return what.endswith('?')

def said_nothing(what):
    return what.strip() == ""
    
def is_yelling(what):
    return what.isupper()

def hey(what):
    if   said_nothing(what): return "Fine. Be that way!"
    elif is_yelling(what):   return "Whoa, chill out!"
    elif is_question(what):  return "Sure."
    else:                    return "Whatever."

File: Exercism.io/elixir/bob/bob.exs
defmodule Teenager do

 defp rules do
    shout?    = fn p -> p == String.upcase(p) and p != String.downcase(p) end
    question? = fn p -> p |> String.ends_with? "?" end
    silence?  = fn p -> p |> String.strip |> String.length == 0 end
    default   = fn _ -> true end

    [{ shout?    , "Whoa, chill out!" },
     { question? , "Sure."},
     { silence?  , "Fine. Be that way!"}, 
     { default   , "Whatever."}]
  end
  
  defp answer(input, [{f,v}|t]) do
    if f.(input), do: v, else: answer(input, t)
  end

  def hey(input) do
    input |> answer(rules)
  end

end


